src/osp/Universe.cpp

```diff
@@ -51,13 +51,13 @@ void Universe::sat_remove(Satellite sat)
 }
 
 
-Vector3s Universe::sat_calc_pos(Satellite referenceFrame, Satellite target)
+Vector3s Universe::sat_calc_pos(Satellite referenceFrame, Satellite target) const
 {
-    auto view = m_registry.view<UCompTransformTraj>();
+    auto const view = m_registry.view<const UCompTransformTraj>();
```

Comment:
Note that .view<const ...> probably has no meaningful effect, unless you're asking it to return a const&, which doesn't seem to be the case here.

Reply:
Views provide access to a registry, and are fairly cheap to construct. They're still internally linked to the registry they were created from. Marking them const assures that they won't modify the registry.

Reply:
Right, but .view<const ...> doesn't necessarily mean anything unless you're making it return a const&, because, e.g.

int foo(void);
const int foo(void);

Can both be assigned to a non-const int.

So you'll need to either make it return a reference, or find a different way to provide immutability.

Reply:
It refuses to compile without the const for UCompTransformTraj.